apps/wallet/src/ui/app/components/nft-display/index.tsx

```diff
@@ -45,20 +45,27 @@ function NFTDisplayCard({
 }: NFTsProps) {
     const { filePath, nftObjectID, nftFields, fileExtensionType, objType } =
         useNFTBasicData(nftobj);
+
+    const { data: originByteNft } = useOriginbyteNft(nftObjectID);
+
     const name = nftFields?.name || nftFields?.metadata?.fields?.name;
     const objIDShort = useMiddleEllipsis(nftObjectID);
     const nftTypeShort = useMiddleEllipsis(
         objType,
         OBJ_TYPE_MAX_LENGTH,
         OBJ_TYPE_MAX_PREFIX_LENGTH
     );
-    const displayTitle = typeof name === 'string' ? name : objIDShort;
+
+    const displayTitle =
+        originByteNft?.fields.name ||
```

Comment:
Does the return from `useOriginbyteNft` _**guarantee**_ that `name` is a string?

Reply:
Yes, it's on the contract level